### PR TITLE
Add more spoofing options

### LINKED ISSUE #874: [Request]: Backend code cleanup
### Problem

Code is messy and inconsistent. I am aware that there are a lot of contributors, but here's a few things that should be reviewed more carefully. 

I AM NO PROFESSIONAL. I AM JUST BEING PICKY. I UNDERSTAND THIS IS ALL PREFERENCE

### Solution

Please review the following

### Random line breaks

image
image

Line breaking was used to make the separators a little more clear. Great! I agree. Why isn't it done in all classes?
image
image

### Formatting inconsistency

Some statements look like this...
image
image
...and others look like this
image
image

This line seems to have a well formatted if statement...
image
...but in the same exact file this happens...
image
...and then THIS later on in a different class
image

### Legibility

Why do this to make it more "readable"
image
When you could just do this
image
Or just don't do it at all?
image

I understand that this is a short and simple function, but why not make break it up and make it look clean?
**Before:**
image
**After:**
image
Compiles exactly the same except now it's not a cluster fuck at first glance

### Reason

Consistency reflects a positive image and cleanliness is always preferred.

### Additional context

One more time in case you can't read
I AM NO PROFESSIONAL. I AM JUST BEING PICKY. I UNDERSTAND THIS IS ALL PREFERENCE

### LINKED ISSUE #876: [Request]: #874 Cont.
### Problem
#874
#874
#874
#874
Cont. because thread was locked due to contributor immaturity (and Loko).

### Grouping

Some code uses groups, and other code doing almost the same thing is not grouped.
image

### ImGui helper functions vs actual ImGui calls 

Someone also mentioned this (partially) in #868
Either make everything a helper func or just remove this junk entirely?
image
image
image

### Solution

#874

### Reason

#874

### Additional context

#874

### LINKED ISSUE #976: [Bug]: UI Scale not implemented properly (Reopen #749)
### Describe the bug

The UI scale slider does not work properly, as showcased in #749

#749 was closed because "It's not really our implementation of it that's wrong because we're literally using ImGui's official scaling methods."

This bothers me because it clearly is the implementation that is incorrect. Other ImGui based menus implement this perfectly fine.

Rocket League's "Bakkesmod" implements this feature using ImGui's official scaling methods, and it functions as expected.
https://github.com/bakkesmodorg

### Steps To Reproduce

#749

### Expected Behavior

UI will scale based on the slider.

### Screenshots and/or Logs

BakkesMod UI scale slider example

https://streamable.com/x37trm

### Store

Steam

### Additional context

 I expect a response such as the one in #859. I get your point. Don't blame functional framework.

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### LINKED ISSUE #1009: [Bug]: Visible Locally option for Invisibility don't work anymore
### Describe the bug

When "Invisible" and "Visible Locally' options are checked, I'm still invisible locally.

### Steps To Reproduce

1. Check Invisible option
2. Check Visible Locally option

### Expected Behavior

Normally I should see my character locally.

### Screenshots and/or Logs

image

### Store

Epic Games

### Additional context

_No response_

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### PR DESCRIPTION
**Pull request by maybegreat48**
* Added blip type spoofing
* Added rank spoofing
* Added K/D ratio spoofing
* Added badsport spoofing
* Added job points spoofing (buggy, may apply to other players)
* Added player model spoofing
* Fixed menu scaling

Fixes #1009 
Fixes #976 
Fixes #876 
Fixes #874 
Possibly _fixes_ #754 

Marking as draft due to the addition of a clang format file. Feel free to suggest possible improvements to it

### CODE REVIEW

**Comment by Primexz** [comment]
I would recommend extending the GH Actions workflows so that we check the code formatting there with clang-format.
This ensures that the developers always push formatted code.

**Comment by xiaoxiao921** [comment]
The GitHub workflow can wait for another PR

This PR is gonna be hard to review (spoofing features) due to it also containing the clang formatting changes

One thing I can say though is that the cmake for GTA v classes point to your fork instead if the Yim one if I'm not mistaken

**Comment by maybegreat48** [comment]
Yeah, will replace it with the upstream version once the GTAV-Classes PR gets merged
[This](https://github.com/YimMenu/YimMenu/pull/1020/commits/143fffca436a28b9102a1133463a8269569cb1d7) is the main commit that added the spoofing features

**Review by xiaoxiao921 — DISMISSED** [review]
Looks good to me but I wonder if there is a better solution for the gui scaling. Having to constantly remember to use the gui scaling variable sounds annoying but I can't think of a better solution

File: src/native_hooks/all_scripts.hpp [hunk 1]
```diff
@@ -103,7 +138,12 @@ namespace big
 			if (g.weapons.interior_weapon && src->get_arg<BOOL>(0) == false)
 				return;
 
-			HUD::HUD_FORCE_WEAPON_WHEEL(src->get_arg<BOOL>(0));
-		}
-	}
+            HUD::HUD_FORCE_WEAPON_WHEEL(src->get_arg<BOOL>(0));
+        }
+
+        void NETWORK_CASINO_CAN_BET(rage::scrNativeCallContext* src)
```
Comment (xiaoxiao921) [inline src/native_hooks/all_scripts.hpp]: Looks like it is duplicated

**Review by Yimura — DISMISSED** [review]
Code is good, I've got so small things I have locally checked out that I'll push this evening.

They're mostly tweaks to the formatter.

**Review by Yimura — APPROVED** [review]

**Pull request merged**